core/unit_test/tools/TestEventCorrectness.hpp

```diff
@@ -280,5 +292,364 @@ TEST(defaultdevicetype, test_streams) {
 }
 
 #endif
+/** FIXME: OpenMPTarget currently has unexpected fences */
+#ifndef KOKKOS_ENABLE_OPENMPTARGET
+TEST(kokkosp, async_deep_copy) {
+  using namespace Kokkos::Test::Tools;
+  listen_tool_events(Config::DisableAll(), Config::EnableFences());
+  Kokkos::View<float*> left("left", 5), right("right", 5);
+
+  auto success = validate_absence(
+      [&]() {
+        Kokkos::deep_copy(Kokkos::DefaultExecutionSpace(), left, right);
+      },
+      [&](BeginFenceEvent begin) {
+        if (begin.deviceID !=
+            Kokkos::DefaultExecutionSpace().impl_instance_id()) {
+          std::stringstream error_message;
+          error_message
+              << "Fence encountered outside of the default instance, default: "
+              << Kokkos::DefaultExecutionSpace().impl_instance_id()
+              << ", encountered " << begin.deviceID << " , fence name "
+              << begin.name;
+          return MatchDiagnostic{true, {error_message.str()}};
+        }
+        return MatchDiagnostic{false};
+      });
+  ASSERT_TRUE(success);
+}
```

Comment:
Shouldn't we unregister hooks at the end of each unit test?

Reply:
I've chosen to just register new hooks at the start of the next one, which starts by unregistering everything. In a mixed environment where some tests don't want callbacks, this would be necessary though